### DISCUSSION TITLE
Physical Nodes

### DISCUSSION THREAD

**Discussion by Augusto12** [coreemu/core#662] [General]
Hi,

I'm trying to use PhysicalNode and Rj45Node types to bring physical devices in my emulation environment. When I try this, I get the error "Can't instantiate abstract class <Physical|Rj45>Node with abstract methods copy_file, create_dir, create_file". How can I use this Node types? How can I bring physical devices to emulation environment?

**Comment by bharnden** [comment]
RJ45 nodes should work as of the latest release 8.1.0, PhysicalNodes are a bit more experimental and their current working condition is unknown. They may be suffering from the same issue that plagued RJ45 at a basic case, which I can look to fix, but that would not guarantee they will actually work, once resolved.

**Reply by Augusto12** [reply]
Thanks for reply, @bharnden .
I was trying with v8.0.0.0. I will try with the latest that you mentioned. About Physical Nodes, I saw in project's docs that this feature is experimental. My interest now is just put physical devices in emulated environment without concern with metrics... Do you believe that this is possible anyway? If yes, How can I do this? I didn't find examples in the project.

**Reply by bharnden** [reply]
An RJ45 node would basically disable an interface on your host to be integrated into the simulation. As long as whatever machines on the other end are addresses and setup for communication, it should work for that purpose.
